dj/cli/compile.py

```diff
@@ -306,11 +313,28 @@ async def index_nodes(  # pylint: disable=too-many-locals
         for future in done:
             node = future.result()
             nodes[node.name] = node
+            if node.type == NodeType.METRIC and node.name in existing_metrics:
+                existing_metrics.pop(node.name)
             finished.add(node.name)
 
+    # remove existing metrics that were not found when indexing current configs
+    if existing_metrics:
```

Comment:
Just a small nit, but the intent here could be made clearer by creating a separate set called `deleted_metrics`, something like:

```python
existing_metrics = {...}
deleted_metrics = set(existing_metrics)

...
for config in configs:
    ...
    for future in done:
        ...
        if node.type == NodeType.METRIC and node.name in existing_metrics:
            deleted_metrics.remove(node.name)
...
if deleted_metrics:
    await asyncio.wait(
        {remove_node(session, existing_metrics[name]) for name in deleted_metrics},
        return_when=asyncio.ALL_COMPLETED,
    )
```

Reply:
Based on your other comment, it sounds like a better name is `deleted_nodes` 😄

Reply:
Right! :)

Reply:
done!